tempesta_fw/http_match.h

```diff
@@ -61,20 +82,41 @@ typedef enum {
 	_TFW_HTTP_MATCH_A_COUNT
 } tfw_http_match_arg_t;
 
+typedef enum {
+	TFW_HTTP_MATCH_ACT_NA = 0,
+	TFW_HTTP_MATCH_ACT_CHAIN,
+	TFW_HTTP_MATCH_ACT_VHOST,
+	TFW_HTTP_MATCH_ACT_MARK,
+	TFW_HTTP_MATCH_ACT_BLOCK,
+	_TFW_HTTP_MATCH_ACT_COUNT
+} tfw_http_rule_act_t;
+
 typedef struct {
 	tfw_http_match_arg_t type;
 	short len; /* Actual amount of memory allocated for the union below. */
 	union {
 		tfw_http_meth_t method;
+		unsigned int num;
 		TfwAddr addr;
 		char str[0];
 	};
 } TfwHttpMatchArg;
 
+typedef struct {
+	tfw_http_rule_act_t type;
+	union {
+		TfwHttpChain *chain;
+		TfwVhost *vhost;
+		unsigned int mark;
+	};
+} TfwHttpAction;
+
 typedef struct {
 	struct list_head	list;
 	tfw_http_match_fld_t	field; /* Field of a HTTP message to compare. */
+	bool			inv;   /* Comparison inversion (inequality) flag.*/
 	tfw_http_match_op_t 	op;    /* Comparison operator. */
+	TfwHttpAction		act;   /* Rule action. */
 	TfwHttpMatchArg 	arg;   /* A value to be compared with the field.
 					  note: the @arg has variable length. */
 } TfwHttpMatchRule;
```

Comment:
Please arrange members of this structure and the nested structures to minimize memory gaps (e.g. integer enum types should be adjacent where possible, probably it has sense to replace the `bool` fields with integer flags).

Reply:
Corrected.